### PR TITLE
test(common): add unit tests for common helpers

### LINKED ISSUE #2739: test(common): implement unit tests for common helpers
## Feature Request / Enhancement

**Short Description**
Create a unit test suite (`common_test.go`) to test core utility functions under `KubeArmor/common/common.go`.

**Is your feature request related to a problem? Please describe the use case.**
The `common` package provides helper functions like `Clone`, `RemoveStringElement`, `ContainsElement`, `MatchesRegex`, comma expanders, and timestamp formatting logic. Currently, there are no unit tests covering this package, making it prone to regressions during future refactorings.

**Describe the solution you'd like**
Add a `common_test.go` file inside `KubeArmor/common/` that implements table-driven Go tests for:
- `Clone`
- `RemoveStringElement`
- `ContainsElement`
- `MatchesRegex`
- `ObjCommaCanBeExpanded` and `ObjCommaExpandFirstDupOthers`
- `CopyFile`
- `GetDateTimeNow`

**Describe alternatives you've considered**
None.

### PR DESCRIPTION
**Pull request by Rucha0901**
**Purpose of PR?**:

Fixes #2739

**Does this PR introduce a breaking change?**
No

**If the changes in this PR are manually verified, list down the scenarios covered:**:
- Created unit tests for the functions: `Clone`, `RemoveStringElement`, `ContainsElement`, `MatchesRegex`, `ObjCommaCanBeExpanded`, `ObjCommaExpandFirstDupOthers`, `CopyFile`, and `GetDateTimeNow`.
- Verified file syntax and structure.

**Additional information for reviewer?** :
N/A

**Checklist:**
- [ ] Bug fix. Fixes #2739 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [x] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: KubeArmor/common/common_test.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+import (
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+	"time"
+)
+
+func TestClone(t *testing.T) {
```
Comment (Aryan-sharma11) [inline KubeArmor/common/common_test.go]: Since this is only a wrapper function, I'm not sure a dedicated unit test adds much value.

File: KubeArmor/common/common_test.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+	if err != nil {
+		t.Fatalf("Failed to read destination file: %v", err)
+	}
+
+	if string(copiedContent) != string(content) {
+		t.Errorf("Expected %s, got %s", content, copiedContent)
+	}
+}
+
+func TestGetDateTimeNow(t *testing.T) {
```
Comment (Aryan-sharma11) [inline KubeArmor/common/common_test.go]: This as well.

**Review by Aryan-sharma11 — COMMENTED** [review]
Most of the functions in `common` package are  wrapper functions and are already exercised indirectly through tests in other packages. Do these standalone unit tests provide any additional value beyond that?

Let's identify the ones that contain meaningful logic or edge cases and create a list of functions that genuinely warrant dedicated unit tests.
